ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala

```diff
@@ -235,75 +234,78 @@ private[indexer] class JdbcIndexer private[indexer] (
     new SubscriptionResourceOwner(readService)
 
   private def handleStateUpdate(
-      offset: Offset,
-      update: Update,
-  )(implicit loggingContext: LoggingContext): Future[Unit] = {
-    lastReceivedRecordTime = update.recordTime.toInstant.toEpochMilli
-
-    logger.trace(update.description)
-
-    metrics.daml.indexer.lastReceivedRecordTime.updateValue(lastReceivedRecordTime)
-    metrics.daml.indexer.lastReceivedOffset.updateValue(offset.toApiString)
-
-    val result = update match {
-      case PartyAddedToParticipant(
-          party,
-          displayName,
-          hostingParticipantId,
-          recordTime,
-          submissionId,
-          ) =>
-        val entry = PartyLedgerEntry.AllocationAccepted(
-          submissionId,
-          recordTime.toInstant,
-          domain.PartyDetails(party, Some(displayName), participantId == hostingParticipantId)
-        )
-        ledgerDao.storePartyEntry(offset, entry)
+      implicit loggingContext: LoggingContext): Flow[(Offset, Update), Unit, NotUsed] =
+    Flow[(Offset, Update)]
+      .wireTap(Sink.foreach[(Offset, Update)] {
+        case (offset, update) =>
+          lastReceivedRecordTime = update.recordTime.toInstant.toEpochMilli
+
+          logger.trace(update.description)
+
+          metrics.daml.indexer.lastReceivedRecordTime.updateValue(lastReceivedRecordTime)
```

Comment:
We should look at the documentation of this metric and make sure it says this is the last offset received from the ledger, and not the last offset written to the index database. The update of the metric is now a few async boundaries away from writing the data to the database.

Reply:
It's on my radar as well, but I would prefer to make these changes iterative so I can understand the impact of each them

Reply:
It is in the same spot (in the execution flow) as before, but separated by some Akka operators. I am not sure how this could have changed the meaning of it. Am I missing something?